assets/data-port/import/data/reducer.js

```diff
@@ -0,0 +1,185 @@
+import {
+	START_FETCH_IMPORT_DATA,
+	SUCCESS_FETCH_IMPORT_DATA,
+	ERROR_FETCH_IMPORT_DATA,
+	START_START_IMPORT,
+	SUCCESS_START_IMPORT,
+	ERROR_START_IMPORT,
+	ERROR_UPLOAD_IMPORT_DATA_FILE,
+	START_UPLOAD_IMPORT_DATA_FILE,
+	SUCCESS_UPLOAD_IMPORT_DATA_FILE,
+} from './constants';
+
+import { merge } from 'lodash';
+
+const DEFAULT_STATE = {
+	isFetching: true,
+	fetchError: false,
+	data: {
+		completedSteps: [],
+		upload: {
+			isSubmitting: false,
+			errorMsg: null,
+			levels: {
+				courses: {
+					isUploaded: false,
+					inProgress: false,
+					hasError: false,
+					errorMsg: null,
+					filename: null,
+				},
+				lessons: {
+					isUploaded: false,
+					inProgress: false,
+					hasError: false,
+					errorMsg: null,
+					filename: null,
+				},
+				questions: {
+					isUploaded: false,
+					inProgress: false,
+					hasError: false,
+					errorMsg: null,
+					filename: null,
+				},
+			},
+		},
+		import: {
+			status: '',
+			percentage: 0,
+		},
+	},
+};
+
+function updateLevelState( state, levelKey, attributes ) {
+	const newState = {
+		...state,
+		data: {
+			...state.data,
+			upload: {
+				...state.data.upload,
+				levels: {
+					...state.data.upload.levels,
```

Comment:
One good practice to have updates like this more sane is to try and keep the state as flat as possible. I think we could get away with ditching the top-level `data` and `upload.levels` here.
On top of that, we could create separate reducers scoped for each step with [combineReducers](https://redux.js.org/api/combinereducers).

Reply:
Good call on flattening. I was kinda combining the original data structure from the uploader and setup wizards.

Reply:
Removed `data` level in https://github.com/Automattic/sensei/pull/3306/commits/6f4c8b1c102a8d02ea191177bd28ce5e269b61bf

Removed `levels` level in https://github.com/Automattic/sensei/pull/3306/commits/605d7615a9366ca5c166540c056c7a8494618f3c